common/types/string.go

```diff
@@ -186,6 +186,10 @@ func (s String) Value() any {
 	return string(s)
 }
 
+func (s String) String() string {
+	return strconv.Quote(string(s))
```

Comment:
I feel like we should leave this one alone and maybe use %q in places where we print the string. Seems less likely to hit a change-detector test issue that way too.

Reply:
Unfortunately we cannot use `%q` because that would quote everything, including ints. We pass ref.Value to `%s`.